src/bridges/erc4626/VaultBridge.sol

```diff
@@ -0,0 +1,163 @@
+     * the RollupProcessor.sol
+     */
+    function convert(
+        AztecTypes.AztecAsset memory inputAssetA,
+        AztecTypes.AztecAsset memory inputAssetB,
+        AztecTypes.AztecAsset memory outputAssetA,
+        AztecTypes.AztecAsset memory outputAssetB,
+        uint256 totalInputValue,
+        uint256 interactionNonce,
+        uint64 auxData,
+        address rollupBeneficiary
+    )
+        external
+        payable
+        override
+        returns (
+            uint256 outputValueA,
+            uint256 outputValueB,
+            bool isAsync
+        )
+    {
+        // // ### INITIALIZATION AND SANITY CHECKS
+        require(msg.sender == rollupProcessor, Unathorized());
+
+        if (testPair(outputAssetA.erc20Address, inputAssetA.erc20Address)) {
+            outputValueA = enter(outputAssetA.erc20Address, inputAssetA.erc20Address, totalInputValue);
+        } else if (testPair(inputAssetA.erc20Address, outputAssetA.erc20Address)) {
+            outputValueA = exit(inputAssetA.erc20Address, outputAssetA.erc20Address, totalInputValue);
+        } else {
+            revert('invalid vault token pair address');
+        }
+    }
+
+    /**
+     * @notice Function which test whether a token is a vaild asset for a given vault
+     * @dev This method is used to check whether input and output assets are matching vault share/asset pairs
+     *
+     * @param a address of erc4626 vault
+     * @param b address of the vault asset token
+     *
+     */
+    function testPair(address a, address b) public returns (bool) {
+        try IERC4626(a).asset() returns (IERC20 token) {
+            return address(token) == b;
+        } catch {
+            return false;
+        }
+    }
+
+    /**
+     * @notice Internal Function used to stake
+     * @dev This method deposits an exact amount of asset into an erc4626 vault then approves the rollup processor
+     *
+     * @param vault address of erc4626 vault
+     * @param token address of the vault asset token
+     * @param amount amount of an asset to be burned
+   
+     */
+    // enter by deposit exact assets
+    function enter(
+        address vault,
+        address token,
+        uint256 amount
+    ) internal returns (uint256) {
+        IERC20 depositToken = IERC20(token);
+        // depositToken.approve(vault, amount);
+        uint256 prev = IERC4626(vault).balanceOf(address(this));
+        IERC4626(vault).deposit(amount, address(this));
+        uint256 _after = IERC4626(vault).balanceOf(address(this));
+        // IERC4626(vault).approve(rollupProcessor, _after.sub(prev));
+        return _after.sub(prev);
+    }
+
+    /**
+     * @notice Internal Function used to unstake
+     * @dev This method redeems an exact number of shares into an erc4626 vault then approves the rollup processor
+     *
+     * @param vault address of erc4626 vault
+     * @param token address of the vault
+     * @param amount amount of shares to be redeemed
+     *
```

Comment:
Missing natspec for the return value

Reply:
@johhonn this was not yet addressed. Please address all our comments before requesting a review.